javatests/arcs/showcase/typevariables/variable.arcs

```diff
@@ -0,0 +1,39 @@
+// An online shop needs a service to process orders. However, the order item SKU is
+// considered sensitive information and needs to be redacted.
+// In this recipe, we demonstrate how one can use type variables to build a redactor
+// particle.
+meta
+  namespace: arcs.showcase.typevariables
+
+// This is an ingress particle. It brings order data into the system.
+particle OrderIngestion in '.OrderIngestion'
+  data: writes [Product {sku: Text, name: Text, price: Number}]
+
+// Type variables are used here to ensure that only need-to-know data is passed to
+// the particle. `SkuRedactor` only has knowledge of the input handle's `sku` field
+// (any schema after `with` clause we call a "constraint"). Yet, all of the rest of
+// the fields will flow through the particle into the output handle. The input and
+// output handles of the same variable name can be of any type as long as
+// constraints are met.
+particle SkuRedactor in '.SkuRedactor'
+  input: reads [~a with {sku: Text}]
+  output: writes [~a]
+  claim output is redactedSku
+
+// The shop processes orders in the consumer service. It has confidence that the sku
+// is redacted due to the check on the data handle.
+particle Consumer in '.Consumer'
+  data: reads [Product {sku: Text, name: Text, price: Number}]
+  check data is redactedSku
```

Comment:
Should be `check data.sku is redactedSku`. 

Otherwise, the check will fail when combined with `claim ouptut.sku is redactedSku` as the other fields won't have any labels.

Reply:
Given my other comment -- how about I add these to the showcase when checks / claims for type variables are ready?